File: Arrays/TripletSum/tripletSum.cpp
#include <iostream>
using std::cin;
using std::cout;
using std::endl;

int findTriplet(int arr[], int n, int x) {
    int ans {};
    for (int i{}; i<n; i++) {
        for (int j{i+1}; j<n; j++) {
            for (int k{j+1}; k<n; k++) {
                if (arr[i] + arr[j] + arr[k] == x) ans++;
            }
        }
    }

    return ans;
}


int main () {
    int testCase {}, n {}, arr[100] {}, x {};
    cin >> testCase;

    while (testCase--) {
        cin >> n;
        for (int i{}; i<n; i++) cin >> arr[i];
        cin >> x;
        int ans {findTriplet(arr, n, x)};
        cout << ans << endl;
    }
}

File: Arrays/TripletSum/tripletSum.js
(function () {
  let arr = [1, 2, 3, 4, 5, 6, 7];
  let x = 12;
  let ans = 0;

  for (let i = 0; i < arr.length; i++) {
    for (let j = i + 1; j < arr.length; j++) {
      for (let k = j + 1; k < arr.length; k++) {
        if (arr[i] + arr[j] + arr[k] === x) ans++;
      }
    }
  }

  console.log(ans);
})();
